src/axom/quest/examples/containment_driver.cpp

```diff
@@ -35,88 +36,230 @@ namespace slic = axom::slic;
 
 using UMesh = mint::UnstructuredMesh<mint::SINGLE_SHAPE>;
 
-using TriVertIndices = primal::Point<axom::IndexType, 3>;
-using SpaceTriangle = primal::Triangle<double, 3>;
+//------------------------------------------------------------------------------
 
-using Octree3D = quest::InOutOctree<3>;
+template <int DIM>
+class ContainmentDriver
+{
+public:
+  using CellVertIndices = primal::Point<axom::IndexType, DIM>;
 
-using GeometricBoundingBox = Octree3D::GeometricBoundingBox;
-using SpacePt = Octree3D::SpacePt;
-using SpaceVector = Octree3D::SpaceVector;
-using GridPt = Octree3D::GridPt;
-using BlockIndex = Octree3D::BlockIndex;
+  using InOutOctreeType = quest::InOutOctree<DIM>;
 
-//------------------------------------------------------------------------------
+  using GeometricBoundingBox = typename InOutOctreeType::GeometricBoundingBox;
+  using SpacePt = typename InOutOctreeType::SpacePt;
+  using SpaceVector = typename InOutOctreeType::SpaceVector;
+  using GridPt = typename InOutOctreeType::GridPt;
+  using BlockIndex = typename InOutOctreeType::BlockIndex;
+  using SpaceCell = typename InOutOctreeType::SpaceCell;
 
-/** Computes the bounding box of the surface mesh */
-GeometricBoundingBox compute_bounds(mint::Mesh* mesh)
-{
-  SLIC_ASSERT(mesh != nullptr);
+// Determine an appropriate execution policy
+#if defined(AXOM_USE_RAJA) && defined(AXOM_USE_OPENMP) && \
+  defined(RAJA_ENABLE_OPENMP)
+  using ExecPolicy = axom::OMP_EXEC;
+#else
+  using ExecPolicy = axom::SEQ_EXEC;
+#endif
 
-  GeometricBoundingBox meshBB;
-  SpacePt pt;
+  ContainmentDriver() = default;
 
-  for(int i = 0; i < mesh->getNumberOfNodes(); ++i)
+  ~ContainmentDriver()
   {
-    mesh->getNode(i, pt.data());
-    meshBB.addPoint(pt);
+    delete m_surfaceMesh;
+    delete m_octree;
   }
 
-  SLIC_ASSERT(meshBB.isValid());
+#ifdef AXOM_USE_C2C
+  void loadContourMesh(const std::string& inputFile, int segmentsPerKnotSpan)
+  {
+    quest::C2CReader* reader = new quest::C2CReader();
```

Comment:
This could just be `quest::C2CReader reader;`. Then you wouldn't have to worry about deleting it later.